src/HTTPRequest/HTTPRequest.cpp

```diff
@@ -0,0 +1,222 @@
+	: _content(request), _keepAlive(true), _contentLength(0)
+	{
+	// try {
+		parse();
+		setup();
+		displayParsedRequest();
+	// } catch {
+		// throw failedHTTPRequest
+	// }
+}
+
+HTTPRequest::~HTTPRequest() {
+	delete parser;
+};
+
+int	HTTPRequest::parse(){
+	// displayRequest();
+	HTTPLexer	lex;
+	lex.tokenize(_content);
+	// lex.displayTokenList();
+
+	parser = new HTTPParser(lex.getTokens());
+	if (parser->parse())
+		return 1;
+	// parser->displayAST();
+	return 0;
+}
+
+int	HTTPRequest::setup(){
+	std::string headers[] = \
+	{ 	"cache-control", "connection", "date", "'pragma", "trailer", \
+		"transfer-encoding", "upgrade", "via", "warning", \
+		"accept", "accept-charset", "accept-encoding", "accept-language", \
+		"authorization", "expect", "from", "host", "if-match", "if-modified-since", \
+		"if-none-match", "if-range", "if-unmodified-since", "max-forwards", \
+		"proxy-authorization", "range", "referer", "referrer-policy", "te", "user-agent", \
+		"allow", "content-encoding", "content-language", "content-length", \
+		"content-location", "content-md5", "content-range", "content-type", \
+		"expires", "last-modified", "extension-header"};
+	std::vector<std::string> validHeaders(headers, headers + sizeof(headers) / sizeof(std::string));
+
+	std::list<Node>::const_iterator it = parser->getNodes().begin();
+
+	for (; it != parser->getNodes().end(); it++) {
+		switch (it->_type) {
+			case Method: {
+				this->_method = it->_content;
+				break ;
+			}
+			case URI: {
+				if (it->_content.find(".py") != std::string::npos) {
+					this->_requestURI = it->_content.substr(0, it->_content.find('?'));
+					extractQuery(it->_content);
+				} else {
+					this->_requestURI = it->_content;
+				}
+				break ;
+			}
+			case Protocol: {
+				this->_protocol = it->_content;
+				break ;
+			}
+			case Name: {
+				std::string paramName = it->_content;
+				std::transform(paramName.begin(), paramName.end(), paramName.begin(), ::tolower);
+				it++;
+				/* The `if` statement is checking if the `paramName` is a valid header name. It does this by
+				checking if the `paramName` is present in the `validHeaders` list, or if it starts with "sec-",
+				or if it starts and ends with a colon. If any of these conditions are true, then the header is
+				considered valid and its content is stored in the `_params` map. */
+				if (std::count(validHeaders.begin(), validHeaders.end(), paramName) || \
+					paramName.find("sec-") == 0 || \
+					(paramName.find_first_of(':') == 0 && paramName.find_last_of(':') == paramName.size() - 1)) {
+					std::string paramContent = it->_content;
+					std::transform(paramContent.begin(), paramContent.end(), paramContent.begin(), ::tolower);
+					checkIfConnection(paramName, paramContent);
+					checkIfAccept(paramName, paramContent);
+					checkContentLength(paramName, paramContent);
+
+					_params[paramName] = paramContent;
+				} else {
+					//change to exception
+					std::string err = "Invalid header: ";
```

Comment:
invalid headers devem lancar excecoes? ou imprimimos mensagem de erro e segue caminho?

Reply:
Add bad_request flag to issue 400 error_code